import/webview/WebView.qml

```diff
@@ -31,19 +31,32 @@ RawWebView {
         webview.loadFrameScript("chrome://embedlite/content/embedhelper.js");
         webview.loadFrameScript("chrome://embedlite/content/SelectAsyncHelper.js");
         webview.addMessageListeners([
-            "embed:linkclicked",
-            "embed:filepicker",
-            "embed:selectasync",
-            "embed:alert",
-            "embed:confirm",
-            "embed:prompt",
-            "embed:auth",
-            "embed:permissions",
-            "embed:login",
-            "Content:ContextMenu"]);
+                                        "embed:linkclicked",
+                                    ]);
+    }
+
+    PickerOpener {
+        id: pickerOpener
+
+        property QtObject pageStackOwner: helper.findParentWithProperty(webView, "pageStack")
+
+        pageStack: pageStackOwner ? pageStackOwner.pageStack : undefined
+        contentItem: webview
+    }
+
+    PopupOpener {
+        id: popupOpener
+
+        pageStack: pickerOpener.pageStack
+        parentItem: webview
+        contentItem: webview
     }
 
     onRecvAsyncMessage: {
+        if (pickerOpener.handlesMessage(message) || popupOpener.handlesMessage(message)) {
```

Comment:
One other possibility: instead of using a Connections object in the pickerOpener/popupOpener, just have a "function recvAsyncMessage(message, data, webview)" method, which this one calls if that one handlesMessage(message).  e.g.:

```
if (pickerOpener.handlesMessage(message)) {
    pickerOpener.recvAsyncMessage(message, data, webview);
} else if (popupOpener.handlesMessage(message) {
    popupOpener.recvAsyncMessage(message, data, webview);
} else {
    if (message == "embed:linkclicked") {
        // ...
    }
}
```

This has the advantage of also ensuring that the symbol resolutions in the pickerOpener / popupOpener are more explicit, since they would reference function parameters rather than the "hidden" signal parameters.

It's not important / it's just a matter of preference, but I thought I'd mention it.  It saves an object allocation too.